src/AudioFileSourceHTTPStream.cpp

```diff
@@ -61,13 +128,58 @@ AudioFileSourceHTTPStream::~AudioFileSourceHTTPStream()
   http.end();
 }
 
+uint32_t AudioFileSourceHTTPStream::readRegular(void *data, uint32_t len, bool nonBlock)
+{
+  return readInternal(data, len, nonBlock);
+}
+
+uint32_t AudioFileSourceHTTPStream::readChunked(void *data, uint32_t len, bool nonBlock)
+{
+  uint32_t bytesRead = 0;
+  uint32_t pos = 0;
+  
+  while(len > 0)
+  {
+    if(len >= next_chunk)
+    {
+      while (next_chunk)
+      {
+        bytesRead = readInternal(data + pos, next_chunk, nonBlock);
+        next_chunk -= bytesRead;
+        pos += bytesRead;
+      }
+      len -= pos;
+      if(!verifyCrlf())
+      {
+        audioLogger->printf("Couldn't read CRLF after chunk, something is wrong !!\n");
+        return 0;
+      }
+      next_chunk = getChunkSize();
+    }
+    else
+    {
+      bytesRead = readInternal(data + pos, len, nonBlock);
+      next_chunk -= bytesRead;
+      len -= bytesRead;
+      pos += bytesRead;
+    }
+    
+  }
+  
+  
+  return pos;
+}
+
 uint32_t AudioFileSourceHTTPStream::read(void *data, uint32_t len)
 {
   if (data==NULL) {
     audioLogger->printf_P(PSTR("ERROR! AudioFileSourceHTTPStream::read passed NULL data\n"));
     return 0;
   }
-  return readInternal(data, len, false);
+  //return readInternal(data, len, false);
```

Comment:
Could you delete the commented out code.  Git history can be used if we need to go back.